src/StringTools.cpp

```diff
@@ -0,0 +1,617 @@
+            [](const Char8 Left, const Char8 Right){
+                return ::tolower(Left) == ::tolower(Right);
+            }
+        );
+    }
+
+    Boole EndsWith(const StringView Str, const StringView Pattern, const Boole CaseSensitive)
+    {
+        size_t StrLen = Str.length();
+        size_t PatternLen = Pattern.length();
+        if( PatternLen > StrLen || PatternLen == 0 )
+            return false;
+
+        StringView End = Str.substr(StrLen - PatternLen,PatternLen);
+        if( CaseSensitive ) {
+            return ( End == Pattern );
+        }
+
+        return std::equal(End.begin(),End.end(),Pattern.begin(),Pattern.end(),
+            [](const Char8 Left, const Char8 Right){
+                return ::tolower(Left) == ::tolower(Right);
+            }
+        );
+    }
+
+    Boole PatternMatch(const StringView Source, const StringView Pattern, const Boole CaseSensitive)
+    {
+        StringView::const_iterator SrcIt = Source.begin();
+        StringView::const_iterator PatIt = Pattern.begin();
+        const StringView::const_iterator SrcEnd = Source.end();
+        const StringView::const_iterator PatEnd = Pattern.end();
+
+        Boole EscapedChar = false;
+        while( PatIt != PatEnd && SrcIt != SrcEnd )
+        {
+            Boole MatchResult = false;
+            if( EscapedChar ) {
+                EscapedChar = false;
+                // If we're escaped, doesn't matter what it is.  Process as normal char.
+                MatchResult = ( (*PatIt) == (*SrcIt) );
+                ++PatIt;
+                ++SrcIt;
+            }else if( (*PatIt) == '*' ) {
+                EscapedChar = false;
+                MatchResult = MatchAsterisk(SrcIt,SrcEnd,PatIt,PatEnd,CaseSensitive);
+            }else if( (*PatIt) == '[' ) {
+                EscapedChar = false;
+                MatchResult = MatchRangeCharacter(SrcIt,SrcEnd,PatIt,PatEnd);
+            }else if( (*PatIt) == '?' ) {
+                EscapedChar = false;
+                MatchResult = MatchQuestionMark(SrcIt,SrcEnd,PatIt,PatEnd);
+            }else if( (*PatIt) == '\\' ) {
+                EscapedChar = true;
+                ++PatIt;
+                continue;
+            }else{
+                EscapedChar = false;
+                if( CaseSensitive ) {
+                    MatchResult = ( (*PatIt) == (*SrcIt) );
+                }else{
+                    MatchResult = CaseInsensitiveCompare(PatIt,SrcIt);
+                }
+                ++PatIt;
+                ++SrcIt;
+            }
+
+            if( !MatchResult ) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Convert-To-Data functions
+
+    Boole ConvertToBool(const StringView ToConvert, const Boole Default)
+    {
+        auto Checker = [ToConvert](const StringView Str) {
+            if( Str.size() == ToConvert.size() ) {
```

Comment:
This would be better as a call to std::equal which likely does a range length check on its own.

Reply:
Given it's parameters, I don't see how it could:
https://en.cppreference.com/w/cpp/algorithm/equal